src/Azure/Azure.Quantum.Client.Test/WorkspaceTest.cs

```diff
@@ -3,246 +3,239 @@
 
-            Assert.IsNotNull(receivedJob.Workspace);
-
-            Assert.AreEqual(
-                expected: jobId,
-                actual: receivedJob.Details.Id);
+            AssertJob(job);
+            Assert.IsTrue(job.Succeeded);
         }
 
         [TestMethod]
-        public void GetJobTest()
+        [Ignore]
+        public async Task GetJobTest()
         {
-            string jobId = Guid.NewGuid().ToString();
-
-            // Craft response
-            SetJobResponseMessage(jobId);
-
-            // Get Job
-            IWorkspace workspace = GetWorkspace();
-
-            CloudJob receivedJob = workspace.GetJob(jobId);
-
-            // Validate request
-            ValidateJobRequestMessage(jobId, HttpMethod.Get);
+            IWorkspace workspace = GetLiveWorkspace();
 
-            // Validate response
-            Assert.IsNotNull(receivedJob);
+            // Since this is a live workspace, we don't have much control about what jobs are in there
+            // Get the jobs, and call Get on the first.
+            await foreach (var job in workspace.ListJobsAsync())
+            {
+                AssertJob(job);
 
-            Assert.IsNotNull(receivedJob.Workspace);
+                var current = workspace.GetJob(job.Id);
+                AssertJob(current);
+                Assert.AreEqual(job.Id, current.Id);
 
-            Assert.AreEqual(
-                expected: jobId,
-                actual: receivedJob.Details.Id);
+                break;
+            }
         }
 
         [TestMethod]
-        public void CancelJobTest()
+        [Ignore]
+        public async Task CancelJobTest()
         {
-            string jobId = Guid.NewGuid().ToString();
-
-            // Craft response
-            SetJobResponseMessage(jobId);
-
-            // Cancel Job
-            IWorkspace workspace = GetWorkspace();
-
-            CloudJob receivedJob = workspace.CancelJob(jobId);
-
-            // Validate request
-            ValidateJobRequestMessage(jobId, HttpMethod.Delete, HttpMethod.Get);
-
-            // Validate response
-            Assert.IsNotNull(receivedJob);
-
-            Assert.IsNotNull(receivedJob.Workspace);
+            // Create Job
+            IWorkspace workspace = GetLiveWorkspace();
 
-            Assert.AreEqual(
-                expected: jobId,
-                actual: receivedJob.Details.Id);
+            var job = await SubmitTestProblem(workspace);
+            AssertJob(job);
 
-            // Convenience method
-            CloudJob job = new CloudJob(workspace, CreateJobDetails(jobId));
-            string newJobId = Guid.NewGuid().ToString();
-            SetJobResponseMessage(newJobId);
+            try
+            {
+                var result = workspace.CancelJob(job.Id);
+                AssertJob(result);
+            }
+            catch (WorkspaceClientException e)
+            {
+                Assert.AreEqual((int)HttpStatusCode.Conflict, e.Status);
```

Comment:
Is this exception expected to happen? If yes, this would be better `Assert.Throws<Exception>(() => SomethingThatThrowsAnException());`